cura/Settings/IntentManager.py

```diff
@@ -0,0 +1,141 @@
+        pass
+
+    ##  This class is a singleton.
+    @classmethod
+    def getInstance(cls):
+        if not cls.__instance:
+            cls.__instance = IntentManager()
+        return cls.__instance
+
+    configurationChanged = pyqtSignal()
+
+    ##  Gets the metadata dictionaries of all intent profiles for a given
+    #   configuration.
+    #
+    #   \param definition_id ID of the printer.
+    #   \param nozzle_name Name of the nozzle.
+    #   \param material_id ID of the material.
+    #   \return A list of metadata dictionaries matching the search criteria, or
+    #   an empty list if nothing was found.
+    def intentMetadatas(self, definition_id: str, nozzle_name: str, material_id: str) -> List[Dict[str, Any]]:
+        registry = CuraApplication.getInstance().getContainerRegistry()
+        return registry.findContainersMetadata(definition = definition_id, variant = nozzle_name, material_id = material_id)
+
+    ##  Collects and returns all intent categories available for the given
+    #   parameters. Note that the 'default' category is always available.
+    #
+    #   \param definition_id ID of the printer.
+    #   \param nozzle_name Name of the nozzle.
+    #   \param material_id ID of the material.
+    #   \return A set of intent category names.
+    def intentCategories(self, definition_id: str, nozzle_id: str, material_id: str) -> List[str]:
+        categories = set()
+        for intent in self.intentMetadatas(definition_id, nozzle_id, material_id):
+            categories.add(intent["intent_category"])
+        categories.add("default") #The "empty" intent is not an actual profile specific to the configuration but we do want it to appear in the categories list.
+        return list(categories)
+
+    ##  List of intents to be displayed in the interface.
+    #
+    #   For the interface this will have to be broken up into the different
+    #   intent categories. That is up to the model there.
+    #
+    #   \return A list of tuples of intent_category and quality_type. The actual
+    #   instance may vary per extruder.
+    def currentAvailableIntents(self) -> List[Tuple[str, str]]:
+        application = CuraApplication.getInstance()
+        global_stack = application.getGlobalContainerStack()
+        if global_stack is None:
+            return [("default", "normal")]
+            # TODO: We now do this (return a default) if the global stack is missing, but not in the code below,
+            #       even though there should always be defaults. The problem then is what to do with the quality_types.
+            #       Currently _also_ inconsistent with 'currentAvailableIntentCategoreis', which _does_ return default.
+        quality_groups = application.getQualityManager().getQualityGroups(global_stack)
+        available_quality_types = {quality_group.quality_type for quality_group in quality_groups.values() if quality_group.node_for_global is not None}
+
+        final_intent_ids = set() #type: Set[str]
```

Comment:
```suggestion
        final_intent_ids = set()  # type: Set[str]
```
Pep wants 2 spaces before it and one space after #